metaflow/plugins/pypi/conda_decorator.py

```diff
@@ -0,0 +1,335 @@
+                )
+            else:
+                # This is a namespace package, we therefore create a bunch of
+                # directories so that we can symlink in those separately, and we will
+                # add those paths to the PYTHONPATH for the interpreter. Note that we
+                # don't symlink to the parent of the package because that could end up
+                # including more stuff we don't want
+                self.addl_paths = []
+                for p in custom_paths:
+                    temp_dir = tempfile.mkdtemp(dir=self.metaflow_dir.name)
+                    os.symlink(p, os.path.join(temp_dir, EXT_PKG))
+                    self.addl_paths.append(temp_dir)
+
+        # # Also install any environment escape overrides directly here to enable
+        # # the escape to work even in non metaflow-created subprocesses
+        # from ..env_escape import generate_trampolines
+        # generate_trampolines(self.metaflow_dir.name)
+
+    def runtime_task_created(
+        self, task_datastore, task_id, split_index, input_paths, is_cloned, ubf_context
+    ):
+        if self.disabled:
+            return
+        self.interpreter = (
+            self.environment.interpreter(self.step)
+            if not any(
+                decorator.name in ["batch", "kubernetes"]
+                for decorator in next(
+                    step for step in self.flow if step.name == self.step
+                ).decorators
+            )
+            else None
+        )
+
+    def task_pre_step(
+        self,
+        step_name,
+        task_datastore,
+        meta,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_retries,
+        ubf_context,
+        inputs,
+    ):
+        if self.disabled:
+            return
+        # Add Python interpreter's parent to the path to ensure that any non-pythonic
+        # dependencies in the virtual environment are visible to the user code.
+        # sys.executable points to the Python interpreter in the virtual environment
+        # since we are already inside the task context.
+        os.environ["PATH"] = os.pathsep.join(
+            filter(
+                None,
+                (
+                    os.path.dirname(os.path.realpath(sys.executable)),
+                    os.environ.get("PATH"),
+                ),
+            )
+        )
+        # TODO: Register metadata
+
+    def runtime_step_cli(
+        self, cli_args, retry_count, max_user_code_retries, ubf_context
+    ):
+        if self.disabled:
+            return
+        # Ensure local installation of Metaflow is visible to user code
+        python_path = self.metaflow_dir.name
+        if self.addl_paths is not None:
+            addl_paths = os.pathsep.join(self.addl_paths)
+            python_path = os.pathsep.join([addl_paths, python_path])
+        cli_args.env["PYTHONPATH"] = python_path
```

Comment:
we never did anything with LD_LIBRARY_PATH in the past - we can ship a follow up if needed

Reply:
I am just mentioning because we ran into a very weird issue that took ages to debug and was due to the machine already having LD_LIBRARY_PATH set. It caused the conda environment to look for those libs there first and was, let's just say, very weird.